Add CI 1.73 release notes. (#9920)
harness/developer-hub: release-notes/continuous-integration.md

:::

### Version 1.73

<!-- 2025-04-01 -->

#### New features and enhancements

- Added support for codebase cloning using commit SHA, supporting both long and short commit SHAs. (CI-13445)

> - The following features are now available for Kubernetes environments:

Author: ```suggestion
- The following features are now available to support multi-line output variables:
```

  - Multiline Output Variables: Now supported in CI steps with special character support (\n, \t, \r, \b), maintaining shell-like behavior.

  - Complete Output Support: Available for both output secrets and output strings.

  - JSON Preservation: JSON can now be passed as-is without minification.

> For details, check out this documentation. (CI-15398)

Author: ```suggestion
For details, check out [this documentation](/docs/continuous-integration/use-ci/run-step-settings). (CI-15398)
```

  - Key improvements include Azure SDK integration, cross-platform certificate handling, and enhanced error management. This update significantly improves the plugin's capability to handle various authentication scenarios in enterprise environments. 
- `plugins/kaniko-ecr:1.10.8`: Added three new flags to enhance the `kaniko-ecr` plugin's image handling capabilities (CI16588):

  - `PLUGIN_PUSH_ONLY`: Enables pushing pre-built image tarball without running a build.

  - `PLUGIN_SOURCE_TAR_PATH`: Used in conjunction with **push-only** mode.

  - `PLUGIN_TAR_PATH`, `PLUGIN_DESTINATION_TAR_PATH`: Provides consistent naming with **source-tar-path**. These additions enable more flexible workflows by allowing separation of build and push operations. 
- `plugins/buildx:1.2.0`: Included support for the new `PLUGIN_BUILDX_OPTIONS` flag, allowing users to pass custom options directly to buildx. (CI-16595)
- Updated the default Docker connector for new accounts to point to GAR instead of using the account-level connector (id: harnessImage). (CI-16845)

#### Fixed issues

> - Resolved an issue where the `harness/plugin-build-tool.json` and `harness/plugin-cache-intel.json` files, used for telemetry purposes in CI to detect the build tools, were being downloaded to the `/harness` directory, causing errors in git operations. The metrics files have been moved to the `tmp` directory to prevent population of the `/harness` directory. (ZD-80915, 81193, CI-16782)

Author: ```suggestion
- Resolved an issue where some Harness-generated files were being downloaded to the `/harness` directory, causing errors in git operations. These files have been moved to the `tmp` directory to prevent population of the `/harness` directory. (ZD-80915, 81193, CI-16782)
```
